src/test/ui/hr-subtype/placeholder-pattern.rs

```diff
@@ -0,0 +1,18 @@
+// check-pass
+// Check that higher ranked subtyping correctly works when using
+// placeholder patterns.
+fn hr_subtype<'c>(f: for<'a, 'b> fn(&'a (), &'b ())) {
+    let _: for<'a> fn(&'a (), &'a ()) = f;
+    let _: for<'a, 'b> fn(&'a (), &'b ()) = f;
+    let _: for<'a> fn(&'a (), &'c ()) = f;
+    let _: fn(&'c (), &'c ()) = f;
```

Comment:
can we add a test like this...

```rust
let (x, _): (u32, fn(...)) = (22, f);
```

...to cover the case where the pattern has some bindings but not all parts are bound (we could add a few more variants too, like `(_, x)` and `(x, y)` and `(..)`.